pkg/smartcontract/parameter_test.go

```diff
@@ -887,12 +887,58 @@ func TestParameterFromValue(t *testing.T) {
 			err:   "invalid i value",
 		},
 		{
-			value: make(map[string]int),
-			err:   "unsupported operation: map[string]int type",
+			value:   make(map[string]int),
+			expType: MapType,
+			expVal:  []ParameterPair{},
 		},
 		{
-			value: []any{1, 2, make(map[string]int)},
-			err:   "unsupported operation: map[string]int type",
+			value:   make(map[string]int),
+			expType: MapType,
+			expVal:  []ParameterPair{},
+		},
+		{
+			value:   map[string]string{"key": "value"},
+			expType: MapType,
+			expVal: []ParameterPair{{
+				Key: Parameter{
+					Type:  StringType,
+					Value: "key",
+				},
+				Value: Parameter{
+					Type:  StringType,
+					Value: "value",
+				},
+			}},
+		},
+		{
+			value:   map[int]int64{42: 100500},
+			expType: MapType,
+			expVal: []ParameterPair{{
+				Key: Parameter{
+					Type:  IntegerType,
+					Value: big.NewInt(42),
+				},
+				Value: Parameter{
+					Type:  IntegerType,
+					Value: big.NewInt(100500),
+				},
+			}},
+		},
+		{
+			value: make(chan int),
+			err:   "unsupported operation: chan int type",
+		},
+		{
+			value: []any{1, 2, make(chan int)},
+			err:   "unsupported operation: chan int type",
+		},
```

Comment:
Technically related to the previous commit, but it's a nitpicking.

Reply:
It's not, see an old test above that pushed an array and ended up in `unsupported operation: map[string]int type`, it's the same with reflection-based arrays, but it changes here to this test effectively since maps work correctly now and we need something more fancy.